### ISSUE TITLE
Delete contact fails with "Reuest failed with status code 500"

### ISSUE THREAD

**Issue by iammattmartin** [chatwoot/chatwoot#3649]
**Describe the bug**

Trying to delete once specific contact, generates a 500 error. All other contacts can be deleted without issue.

```
Dec 24 14:28:22 messenger chatwoot-web.1[315263]: I, [2021-12-24T14:28:22.563412 #315263]  INFO -- : source=rack-timeout id=0c9e4165-3a83-4c19-b113-879dad2d1588 timeout=15000ms state=ready
Dec 24 14:28:22 messenger chatwoot-web.1[315263]: I, [2021-12-24T14:28:22.566127 #315263]  INFO -- : [0c9e4165-3a83-4c19-b113-879dad2d1588] Started DELETE "/api/v1/accounts/2/contacts/98395" for x.x.x.x at 2021-12-24 14:28:22 +0000
Dec 24 14:28:22 messenger chatwoot-web.1[315263]: I, [2021-12-24T14:28:22.569717 #315263]  INFO -- : [0c9e4165-3a83-4c19-b113-879dad2d1588] Processing by Api::V1::Accounts::ContactsController#destroy as JSON
Dec 24 14:28:22 messenger chatwoot-web.1[315263]: I, [2021-12-24T14:28:22.570236 #315263]  INFO -- : [0c9e4165-3a83-4c19-b113-879dad2d1588]   Parameters: {"account_id"=>"2", "id"=>"98395"}
Dec 24 14:28:22 messenger chatwoot-web.1[315263]: I, [2021-12-24T14:28:22.625550 #315263]  INFO -- : [0c9e4165-3a83-4c19-b113-879dad2d1588] Completed 500 Internal Server Error in 55ms (ActiveRecord: 16.1ms | Allocations: 5070)
Dec 24 14:28:22 messenger chatwoot-web.1[315263]: F, [2021-12-24T14:28:22.629574 #315263] FATAL -- : [0c9e4165-3a83-4c19-b113-879dad2d1588]
Dec 24 14:28:22 messenger chatwoot-web.1[315263]: [0c9e4165-3a83-4c19-b113-879dad2d1588] ActiveRecord::InvalidForeignKey (PG::ForeignKeyViolation: ERROR:  update or delete on table "contacts" violates foreign key constraint "fk_rails_80b30e3459" on table "csat_survey_responses"
Dec 24 14:28:22 messenger chatwoot-web.1[315263]: DETAIL:  Key (id)=(98395) is still referenced from table "csat_survey_responses".
Dec 24 14:28:22 messenger chatwoot-web.1[315263]: ):
Dec 24 14:28:22 messenger chatwoot-web.1[315263]: [0c9e4165-3a83-4c19-b113-879dad2d1588]
Dec 24 14:28:22 messenger chatwoot-web.1[315263]: [0c9e4165-3a83-4c19-b113-879dad2d1588] app/controllers/api/v1/accounts/contacts_controller.rb:94:in `destroy'
Dec 24 14:28:22 messenger chatwoot-web.1[315263]: [0c9e4165-3a83-4c19-b113-879dad2d1588] app/controllers/concerns/switch_locale.rb:24:in `set_locale'
Dec 24 14:28:22 messenger chatwoot-web.1[315263]: [0c9e4165-3a83-4c19-b113-879dad2d1588] app/controllers/concerns/switch_locale.rb:16:in `switch_locale_using_account_locale'
Dec 24 14:28:22 messenger chatwoot-web.1[315263]: [0c9e4165-3a83-4c19-b113-879dad2d1588] app/controllers/concerns/request_exception_handler.rb:11:in `handle_with_exception'
Dec 24 14:28:22 messenger chatwoot-web.1[315263]: [0c9e4165-3a83-4c19-b113-879dad2d1588] app/controllers/concerns/switch_locale.rb:24:in `set_locale'
Dec 24 14:28:22 messenger chatwoot-web.1[315263]: [0c9e4165-3a83-4c19-b113-879dad2d1588] app/controllers/concerns/switch_locale.rb:11:in `switch_locale'
Dec 24 14:28:22 messenger chatwoot-web.1[315263]: I, [2021-12-24T14:28:22.631192 #315263]  INFO -- : source=rack-timeout id=0c9e4165-3a83-4c19-b113-879dad2d1588 timeout=15000ms service=68ms state=completed
```

**Comment by iammattmartin** [comment]
This is the only contact with a CSAT (testing), so it appears it needs to delete the CSAT before trying to delete the contact.

**Comment by pranavrajs** [comment]
@iammattmartin It seems like the association for CSAT is blocking the contact deletion. 

@muhsin-k Didn't we encounter this before? I remember discussing this relationship. Can you please confirm if this issue is already fixed or not?

**Comment by muhsin-k** [comment]
@pranavrajs I am able to delete contacts that have CSAT survey responses in one of his conversation.

**Comment by muhsin-k** [comment]
@iammattmartin Could you please share your chatwoot version? Seems like the issue already fixed in [v2.1](https://github.com/chatwoot/chatwoot/releases/tag/v2.0.1). We have added some migrations for clearing foreign key associations, which are preventing deletions (https://github.com/chatwoot/chatwoot/pull/3613)

**Comment by iammattmartin** [comment]
Upgrading has resolved this. It was 2.0.
